elasticgraph-warehouse/spec/unit/elastic_graph/warehouse/schema_definition/results_extension_spec.rb

```diff
@@ -0,0 +1,202 @@
+          end
+
+          config = results.warehouse_config
+
+          expect(config).to eq({"tables" => {}})
+        end
+
+        it "excludes object types without warehouse_table definitions" do
+          results = define_warehouse_schema do |s|
+            s.object_type "Product" do |t|
+              t.field "id", "ID"
+              t.field "name", "String"
+              t.warehouse_table "products"
+            end
+
+            s.object_type "Category" do |t|
+              t.field "id", "ID"
+              t.field "name", "String"
+              # No warehouse_table definition
+            end
+          end
+
+          config = results.warehouse_config
+
+          expect(config["tables"].keys).to contain_exactly("products")
+          expect(config["tables"]).not_to have_key("Category")
+        end
+
+        it "sorts warehouse tables by name" do
+          results = define_warehouse_schema do |s|
+            s.object_type "Zebra" do |t|
+              t.field "id", "ID"
+              t.warehouse_table "zebras"
+            end
+
+            s.object_type "Apple" do |t|
+              t.field "id", "ID"
+              t.warehouse_table "apples"
+            end
+
+            s.object_type "Mango" do |t|
+              t.field "id", "ID"
+              t.warehouse_table "mangos"
+            end
+          end
+
+          config = results.warehouse_config
+
+          expect(config["tables"].keys).to eq(["apples", "mangos", "zebras"])
+        end
+
+        it "memoizes the warehouse config" do
+          results = define_warehouse_schema do |s|
+            s.object_type "Product" do |t|
+              t.field "id", "ID"
+              t.warehouse_table "products"
+            end
+          end
+
+          first_call = results.warehouse_config
+          second_call = results.warehouse_config
+
+          expect(first_call).to be(second_call)
+        end
+
+        it "handles types with complex field types" do
+          results = define_warehouse_schema do |s|
+            s.object_type "Event" do |t|
+              t.field "id", "ID"
+              t.field "name", "String"
+              t.field "tags", "[String!]!"
+              t.field "location", "GeoLocation"
+              t.warehouse_table "events"
+            end
+          end
+
+          config = results.warehouse_config
+
+          expect(config["tables"]["events"]["table_schema"]).to eq(<<~SQL.strip)
+            CREATE TABLE IF NOT EXISTS events (
+              id STRING,
+              name STRING,
+              tags ARRAY<STRING>,
+              location STRUCT<lat DOUBLE, lon DOUBLE>
+            )
+          SQL
+        end
+
+        it "handles interface types with warehouse_table definitions" do
+          results = define_warehouse_schema do |s|
+            s.interface_type "Auditable" do |t|
+              t.field "id", "ID"
+              t.field "created_at", "DateTime"
+              t.warehouse_table "auditables"
```

Comment:
Do we consider it valid to define `warehouse_table` on a type which isn't an indexed type?  I guess this applies to all your examples here (and maybe other tests, too...) but I wasn't sure about that.